crates/sui-graphql-rpc/src/types/epoch.rs

```diff
@@ -151,6 +167,32 @@ impl Epoch {
         Ok(SystemStateSummary { native: state })
     }
 
+    /// A commitment by the committee at the end of epoch on the contents of the live object set at
+    /// that time. This can be used to verify state snapshots.
+    async fn live_object_set_digest(&self) -> Result<Option<String>> {
+        self.stored
+            .epoch_commitments
+            .as_ref()
+            .map(|commitments| {
+                bcs::from_bytes::<Vec<EpochCommitment>>(commitments)
+                    .map_err(|e| {
+                        Error::Internal(format!("Error deserializing commitments: {e}")).extend()
+                    })
+                    .map(|commitments| {
+                        commitments
+                            .into_iter()
+                            .map(|commitment| {
+                                let EpochCommitment::ECMHLiveObjectSetDigest(digest) = commitment;
+                                Base58::encode(digest.digest.into_inner())
+                            })
+                            .next()
+                    })
+                    .transpose()
+            })
+            .flatten()
+            .transpose()
```

Comment:
I think `let ... else` would help here.

```suggestion
        let Some(commitments) = &self.stored.epoch_commitments else {
            return Ok(None);
        };
        
        let commitments: Vec<EpochCommitment> = bcs::from_bytes(commitments)
            .map_err(|e| Error::Internal(format!("Error deserializing commitments: {e}")))
            .extend()?;
            
        let digest = commitments
            .into_iter()
            .map(|commitment| {
                let EpochCommitment::ECMHLiveObjectSetDigest(digest) = commitment;
                Base58::encode(digest.digest.into_inner())
            })
            .next();

        Ok(digest)
```